### PR TITLE
[CHORE] clean up object.assign patterns and remove unneeded code comment

### PR DESCRIPTION
**Pull request by runspired**

### CODE REVIEW

File: packages/store/addon/-private/system/core-store.ts [hunk 1]
```diff
@@ -114,21 +114,6 @@ function deprecateTestRegistration(factoryType: 'serializer' | 'adapter', factor
   );
 }
 
-// Implementors Note:
```
Comment (rwjblue) [inline packages/store/addon/-private/system/core-store.ts]: This seems unrelated? Is it just outdated / needing cleanup?

Comment (runspired) [reply packages/store/addon/-private/system/core-store.ts]: jsut outdated
